### PR TITLE
feat: nervos to ethereum monitor

### PR DESCRIPTION
**Pull request by pygman**

### CODE REVIEW

File: offchain-modules/packages/app-monitor/src/monitor.ts [hunk 1]
```diff
@@ -140,6 +177,55 @@ export class Monitor {
     return false;
   }
 
+  async compareCkbLockAndEthMint(lock: CkbLockRecord, mint: EthMintRecord): Promise<boolean> {
+    const checker = (lock: CkbLockRecord, mint: EthMintRecord): string => {
+      if (lock.recipient.toLowerCase() !== mint.recipient.toLowerCase()) {
+        return `lock.recipient:${lock.recipient} !== mint.recipient:${mint.recipient}`;
+      }
+      return '';
+    };
```
Comment (mkxbl) [inline offchain-modules/packages/app-monitor/src/monitor.ts:186]: Add amount check，lockAmount should be gte mintAmount. Also add amount check to `compareEthLockAndCkbMint`

File: offchain-modules/packages/reconc/src/index.ts [hunk 2]
```diff
@@ -58,6 +58,39 @@ export type CkbMintRecord = ToRecord & {
   blockHash?: string;
 };
 
+export type CkbLockRecord = FromRecord & {
+  sender: string;
+  mintId: string;
+  token: string;
+  recipient: string;
+  sudtExtraData: string;
```
Comment (mkxbl) [inline offchain-modules/packages/reconc/src/index.ts:66]:
```suggestion
```

File: offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts [hunk 3]
```diff
@@ -210,4 +225,116 @@ export class CKBRecordObservable {
...
+          firstOutputLockscript.args === committeeMultisigLockscript.args
+        );
+      }),
+      mergeMap(async (ckbTxInfo) => {
+        // tx.objects.map((o) => o.tx_hash)
+        const transaction = ckbTxInfo.tx.transaction;
+        const lockTx = await parseLockTx(transaction);
+        const { amount, recipientAddress, assetIdent, senderAddress } = lockTx!;
+        return {
+          mintId: `${transaction.hash}-${ckbTxInfo.info.tx_index}`,
```
Comment (mkxbl) [inline offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts:268]: Seems not match with EthMintEvent

File: offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts [hunk 4]
```diff
@@ -210,4 +225,116 @@ export class CKBRecordObservable {
...
+      mergeMap((tx) => tx.objects),
+      mergeMap(async (o) => {
+        const txWithStatus = await rpc.getTransaction(o.tx_hash);
+        return {
+          info: o,
+          tx: txWithStatus,
+        };
+      }),
+      rxFilter((ckbTxInfo) => {
+        const firstOutputLockscript = ckbTxInfo.tx.transaction.outputs[0].lock;
```
Comment (mkxbl) [inline offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts:308]: inputs[0].lock === committeeLock for unlockTx

File: offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts (hunk 3 above)
Comment (pygman) [reply offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts:268]: removed wrong mintId

File: offchain-modules/packages/x/src/ckb/model/nervos-asset.ts [hunk 5]
```diff
@@ -9,6 +9,13 @@ export class NervosAsset {
     this.typescriptHash = typescriptHash;
   }
 
+  public static fromToken(token: string): NervosAsset {
```
[the remark this answers is not in the document]
Comment (pygman) [reply offchain-modules/packages/x/src/ckb/model/nervos-asset.ts:12]: renamed

File: offchain-modules/packages/app-monitor/src/monitor.ts (hunk 1 above)
Comment (pygman) [reply offchain-modules/packages/app-monitor/src/monitor.ts:186]: added

File: offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts (hunk 4 above)
Comment (pygman) [reply offchain-modules/packages/x/src/reconc/CKBRecordObservable.ts:308]: getTransactions by committeeLock searchKey to get all lock txs and unlock txs, distinguish lock tx or unlock tx according to whether the first output is equal to committeeLock

**Pull request merged**